.github/scripts/common.py

```diff
@@ -134,7 +148,8 @@ def change_workflow_instance_states(gh_token, tag_value, state_change, dryrun=Fa
 
     elif state_change == 'terminate':
         print("Terminating instances: {}".format(", ".join(instance_ids)))
-        deregister_runner_if_exists(gh_token, tag_value)
+        deregister_offline_runners(gh_token)
```

Comment:
I know you are trying to be conservative here, but unless this is necessary (which seems like a bug), it's probably best if a workflow only touches its own runners. 

You can add more stuff to the periodic cleanup workflow for stuff you want to be workflow global.

Reply:
SGTM. I'll add this to the "cull instances" workflow"